src/components/blog/PaginationControls.tsx

```diff
@@ -0,0 +1,125 @@
+} from "@/components/ui/pagination";
+import { useMemo } from "react";
+
+interface PaginationControlsProps {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+}
+
+const ADJACENT_PAGES = 1;
+
+const PaginationControls = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+}: PaginationControlsProps) => {
+  const paginationItems = useMemo(() => {
+    const items = [];
+    const showEllipsisStart = currentPage > ADJACENT_PAGES + 2;
+    const showEllipsisEnd = currentPage < totalPages - (ADJACENT_PAGES + 1);
+
+    // First page
+    items.push(
+      <PaginationItem key={1}>
+        <PaginationLink
+          onClick={() => onPageChange(1)}
+          isActive={currentPage === 1}
+        >
+          1
+        </PaginationLink>
+      </PaginationItem>,
+    );
+
+    // Start ellipsis
+    if (showEllipsisStart) {
+      items.push(
+        <PaginationItem key="ellipsis-start">
+          <PaginationEllipsis />
+        </PaginationItem>,
+      );
+    }
+
+    // Middle pages
+    for (
+      let i = Math.max(2, currentPage - ADJACENT_PAGES);
+      i <= Math.min(totalPages - 1, currentPage + ADJACENT_PAGES);
+      i++
+    ) {
+      if (i === 1 || i === totalPages) continue;
+      items.push(
+        <PaginationItem key={i}>
+          <PaginationLink
+            onClick={() => onPageChange(i)}
+            isActive={currentPage === i}
+          >
+            {i}
+          </PaginationLink>
+        </PaginationItem>,
+      );
+    }
+
+    // End ellipsis
+    if (showEllipsisEnd) {
+      items.push(
+        <PaginationItem key="ellipsis-end">
+          <PaginationEllipsis />
+        </PaginationItem>,
+      );
+    }
+
+    // Last page
+    if (totalPages > 1) {
+      items.push(
+        <PaginationItem key={totalPages}>
+          <PaginationLink
+            onClick={() => onPageChange(totalPages)}
+            isActive={currentPage === totalPages}
+          >
+            {totalPages}
+          </PaginationLink>
+        </PaginationItem>,
+      );
+    }
+
+    return items;
+  }, [currentPage, totalPages, onPageChange]);
+
+  return (
+    <Pagination>
+      <PaginationContent>
+        <PaginationItem>
+          <PaginationPrevious
+            onClick={() => onPageChange(Math.max(currentPage - 1, 1))}
+            className={
+              currentPage <= 1
+                ? "pointer-events-none opacity-50"
+                : "cursor-pointer"
+            }
+          />
+        </PaginationItem>
+        {paginationItems}
+        <PaginationItem>
+          <PaginationNext
+            onClick={() => onPageChange(Math.min(currentPage + 1, totalPages))}
+            className={
+              currentPage >= totalPages
+                ? "pointer-events-none opacity-50"
+                : "cursor-pointer"
+            }
+          />
+        </PaginationItem>
+      </PaginationContent>
+    </Pagination>
+  );
+};
+
+export default PaginationControls;
```

Comment:
use file naming kebab case rather than pascal case, and adjust the pagination design to the existing design in Figma